stock_move_line_qty_done_constraint/models/stock_move_line.py

```diff
@@ -0,0 +1,31 @@
+# Copyright 2020 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import _, api, models
+from odoo.exceptions import ValidationError
+from odoo.tools import float_compare
+
+
+class StockMoveLine(models.Model):
+
+    _inherit = "stock.move.line"
+
+    @api.constrains("product_qty", "qty_done")
+    def _check_qty_done(self):
+        for rec in self.filtered("picking_id.picking_type_id.qty_done_constraint"):
```

Comment:
```suggestion
        for rec in self.filtered("move_id.picking_type_id.qty_done_constraint"):
```
Could be a bit more generic to make it like this if this is a chance of this to be applied on move lines from manufacturing orders?